src/plugins/discourse/weights.js

```diff
@@ -87,3 +104,30 @@ export function _trustLevelWeight({trustLevel}: User): NodeWeight {
   }
   return weight;
 }
+
+export function _categoryWeight(
+  category: DiscourseCategoryId,
+  {categoryWeights = new Map(), defaultCategoryWeight = 1}: WeightsConfig
```

Comment:
Let's just take a WeightsConfig without any default initialization, let the DiscourseConfig have responsibility for always having a sensible default.